### PR TITLE
Have sys health api always return even in an error case

### LINKED ISSUE #3085: api: sys.health should decode the response body even when the status code is not 200.
According to https://www.vaultproject.io/api/system/health.html, the health api will return meaningful response body in JSON format even if the status code is non-200 (for example 429).

But the vault client implementation returns too early when it thinks there is an error only by examining the status code. The expected behavior is to continue to decode the body and return 

```
{
  "cluster_id": "c9abceea-4f46-4dab-a688-5ce55f89e228",
  "cluster_name": "vault-cluster-5515c810",
  "version": "0.6.2",
  "server_time_utc": 1469555798,
  "standby": false,
  "sealed": false,
  "initialized": true
}
```

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #3085

### CODE REVIEW

**Comment by hongchaodeng** [comment]
I'm getting EOF error with this patch.

File: api/sys_health.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package api
 func (c *Sys) Health() (*HealthResponse, error) {
 	r := c.c.NewRequest("GET", "/v1/sys/health")
 	resp, err := c.c.RawRequest(r)
-	if err != nil {
+	if resp == nil || resp.Body == nil {
```
Comment (xiang90) [inline api/sys_health.go]: the body might already be read by c.rawrequest? we are seeing EOF.

**Comment by jefferai** [comment]
New push should work.

**Comment by hongchaodeng** [comment]
Verified that both "initialized" and "sealed" status are working.

**Comment by xiang90** [comment]
Thanks

**Comment by Saqibm128** [comment]
idk how github works, lol

**Comment by Saqibm128** [comment]
though i do need this too 👍

**Pull request merged**